superchain/superchain.go

```diff
@@ -510,7 +517,9 @@ func (c *ChainConfig) IsEcotone() bool {
 
 var Superchains = map[string]*Superchain{}
 
-var OPChains = map[uint64]*ChainConfig{}
+var OPChains = map[uint64]*ChainConfig{}       // All Chains
+var StandardChains = map[uint64]*ChainConfig{} // Standard Chains
+var FrontierChains = map[uint64]*ChainConfig{} // Frontier Chains
```

Comment:
Existing validation checks run against OPChains (everything). Once this PR is merged, we can follow up with an overhaul of the validation checks which can exclude frontier chains from certain checks.

An alternative here is to not introduce the new mappings, and tests can inspect the new `ChainConfig.Type` property to decide whether a test should apply or not.

Reply:
I'm also happy to address this as a part of this PR, at the reviewer's request.

Reply:
> An alternative here is to not introduce the new mappings, and tests can inspect the new ChainConfig.Type property to decide whether a test should apply or not.

Just an opinion on implementation detail: This feels cleaner than separate mappings, and is easier to expand to more levels. We can make this an enum with values corresponding to the levels [here](https://docs.google.com/presentation/d/1IBHw_KR96DgRaGCvxMSlqMMA3FshbYap9jTQFBO6SXo/edit#slide=id.g269a303f9b1_0_950), then checks can check if a chain's level is e.g. `>=2` to run checks

Reply:
Yep sounds good to me.

Reply:
[8835e78](https://github.com/ethereum-optimism/superchain-registry/pull/140/commits/8835e7891688668ef4b1d411fde1dcd778b69d1c)

Reply:
> if a [frontier] chain gets added to the registry right now then we'll automatically get "support" for that chain in the op-node via the `--network` CLI flag. However, this flag will not actually work when used. This can introduce confusion and is added complexity for little benefit.

From @mslipper in [this doc](https://docs.google.com/document/d/1edDD-ECjV90ltVspZ4RYkiKtiUQqRmySjY-rK1DYp60/edit#heading=h.9sp07hoo4dwm). Let's make sure we don't run into this problem, possibly by making downstream changes to op-node as we bring in these changes.

Alternatively, we could consider having a stronger separation between standard and frontier chains in the Go module, such that frontier chains never actually make it into the downstream modules like op-node.

Reply:
#151